sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/json/JsonConverter.java

```diff
@@ -0,0 +1,378 @@
+  }
+
+  /**
+   * Converts a List of rows into a Jackson JsonNode.
+   *
+   * @param rows The List of RowWrapper objects
+   * @param writeAllPathsAsObjects `true` if array paths should be written as JSON objects instead
+   * @param numericBooleans `true` if booleans should be treated as numbers
+   * @return the JSON representation of the data
+   */
+  public JsonNode convertToJsonDoc(
+      List<RowWrapper> rows, boolean writeAllPathsAsObjects, boolean numericBooleans) {
+    return convertToJsonDoc(
+        rows, new NoOpDeadLeafCollector(), writeAllPathsAsObjects, numericBooleans);
+  }
+
+  /**
+   * Converts a List of rows into a Jackson JsonNode.
+   *
+   * @param rows The List of RowWrapper objects
+   * @param collector a DeadLeafCollector that can be used to collect unreachable information that
+   *     can be deleted.
+   * @param writeAllPathsAsObjects `true` if array paths should be written as JSON objects instead
+   * @param numericBooleans `true` if booleans should be treated as numbers
+   * @return the JSON representation of the data
+   */
+  public JsonNode convertToJsonDoc(
+      List<RowWrapper> rows,
+      DeadLeafCollector collector,
+      boolean writeAllPathsAsObjects,
+      boolean numericBooleans) {
+    int maxDepth = docsProperties.maxDepth();
+    return convertToJsonDoc(rows, collector, writeAllPathsAsObjects, numericBooleans, maxDepth);
+  }
+
+  /**
+   * Takes a List of rows from a documents table, iterating over it and constructing a JSON object
+   * (or array) that is represented by those rows. Note: Because the system avoids
+   * read-before-write, it is possible for the same "path" in a document to contain two different
+   * sets of data from two different versions of the document. When this is encountered during the
+   * conversion, the version that has the latest writetime is accepted, and the older version is
+   * collected in the DeadLeafCollector to be purged in the background.
+   *
+   * @param rows RowWrapper objects
+   * @param collector a DeadLeafCollector implementation, to handle deletion of old and conflicting
+   *     data
+   * @param writeAllPathsAsObjects Instead of writing arrays such as [1, 2], write an object such as
+   *     {"0": 1, "1": 2}
+   * @param numericBooleans If these rows do not support boolean values and are using tinyint
+   *     instead
+   * @param maxDepth The rows' max depth (i.e. the highest pN value)
+   * @return the JsonNode represented by converting the @param rows
+   */
+  private JsonNode convertToJsonDoc(
+      List<RowWrapper> rows,
+      DeadLeafCollector collector,
+      boolean writeAllPathsAsObjects,
+      boolean numericBooleans,
+      int maxDepth) {
+    JsonNode doc = mapper.createObjectNode();
+    Map<String, Long> pathWriteTimes = new HashMap<>();
+    if (rows.isEmpty()) {
+      return doc;
+    }
+
+    for (RowWrapper row : rows) {
+      Long rowWriteTime = row.getLong("writetime(leaf)");
```

Comment:
Mentioned before `"writetime(leaf)"` should be existing in the `TableProperties` as the default method.